validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/ValidatorRestApi.java

```diff
@@ -32,40 +32,46 @@
-        .corsAllowedOrigins(config.getRestApiCorsAllowedOrigins())
-        .hostAllowlist(config.getRestApiHostAllowlist())
-        .exceptionHandler(
-            ServiceUnavailableException.class,
-            (throwable, url) ->
-                new HttpErrorResponse(SC_SERVICE_UNAVAILABLE, "Service unavailable"))
-        .exceptionHandler(
-            BadRequestException.class,
-            (throwable, url) -> new HttpErrorResponse(SC_BAD_REQUEST, throwable.getMessage()))
-        .exceptionHandler(
-            JsonProcessingException.class,
-            (throwable, url) -> new HttpErrorResponse(SC_BAD_REQUEST, throwable.getMessage()))
-        .endpoint(new GetKeys(keyManager))
-        .endpoint(new DeleteKeys(keyManager))
-        .endpoint(new PostKeys(keyManager))
-        .sslCertificate(config.getRestApiKeystoreFile(), config.getRestApiKeystorePasswordFile())
-        .passwordFilePath(
-            ValidatorClientService.getKeyManagerPath(keyManager.getDataDirLayout())
-                .resolve("validator-api-bearer"))
-        .build();
+    RestApiBuilder builder =
+        new RestApiBuilder()
+            .openApiInfo(
+                openApi ->
+                    openApi
+                        .title(
+                            StringUtils.capitalize(
+                                VersionProvider.CLIENT_IDENTITY + " Validator Rest API"))
+                        .version(VersionProvider.IMPLEMENTATION_VERSION)
+                        .bearerAuth(true)
+                        .description("An implementation of the key management standard Rest API.")
+                        .license("Apache 2.0", "https://www.apache.org/licenses/LICENSE-2.0.html"))
+            .openApiDocsEnabled(config.isRestApiDocsEnabled())
+            .listenAddress(config.getRestApiInterface())
+            .port(config.getRestApiPort())
+            .maxUrlLength(config.getMaxUrlLength())
+            .corsAllowedOrigins(config.getRestApiCorsAllowedOrigins())
+            .hostAllowlist(config.getRestApiHostAllowlist())
+            .exceptionHandler(
+                ServiceUnavailableException.class,
+                (throwable, url) ->
+                    new HttpErrorResponse(SC_SERVICE_UNAVAILABLE, "Service unavailable"))
+            .exceptionHandler(
+                BadRequestException.class,
+                (throwable, url) -> new HttpErrorResponse(SC_BAD_REQUEST, throwable.getMessage()))
+            .exceptionHandler(
+                JsonProcessingException.class,
+                (throwable, url) -> new HttpErrorResponse(SC_BAD_REQUEST, throwable.getMessage()))
+            .endpoint(new GetKeys(keyManager))
+            .endpoint(new DeleteKeys(keyManager))
+            .endpoint(new PostKeys(keyManager))
+            .sslCertificate(
+                config.getRestApiKeystoreFile(), config.getRestApiKeystorePasswordFile());
+    // when generating rest-api docs, the keyManager may not be set
```

Comment:
could potentially create a keymanager for use here (in debug-tools fn), but we really don't need to touch keys when we just want docs.. thoughts @ajsutton ?

Reply:
I'd be tempted to have a `NoOpKeyManager` - nulls are never any fun.  Ideally this subcommand wouldn't actually write the bearer token to a file but it probably doesn't matter much if it does - it's only intended to run in CI and is better than having to start the full teku...

Can we only write the bearer file when the API is actually started rather than on construction or does that make life too difficult?

Reply:
it's just locating the files that this is doing, I think its not written till startup but i'll have a play first thing in the morning...